javascript/ql/lib/semmle/javascript/NPM.qll

```diff
@@ -180,6 +180,35 @@ class PackageJson extends JsonObject {
   Module getMainModule() {
     result = min(Module m, int prio | m.getFile() = resolveMainModule(this, prio) | m order by prio)
   }
+
+  /**
+   * Gets the `types` or `typings` field of this package.
+   */
+  string getTypings() { result = this.getPropStringValue(["types", "typings"]) }
```

Comment:
There is also the `"types`"  field under `"exports"`: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-rc/#package-json-exports-imports-and-self-referencing  

(Might not be that relevant, so you can probably skip it).

Reply:
The new `exports`/`imports` fields are too complex to handle in this PR